UDPBroadcast/UDPBroadcastConnection.swift

```diff
@@ -189,27 +189,30 @@ open class UDPBroadcastConnection {
             
             guard sent > 0 else {
                 if let errorString = String(validatingUTF8: strerror(errno)) {
-                    print("UDP connection failed to send data: \(errorString)")
+                    debugPrint("UDP connection failed to send data: \(errorString)")
                 }
                 closeConnection()
                 return
             }
             
             if sent == broadcastMessageLength {
                 // Success
-                print("UDP connection sent \(broadcastMessageLength) bytes")
+                debugPrint("UDP connection sent \(broadcastMessageLength) bytes")
             }
         }
     }
     
     /**
      Close the connection.
      */
-    open func closeConnection() {
+    open func closeConnection(_ reopen: Bool = true) {
         if let source = responseSource {
             source.cancel()
             responseSource = nil
         }
+        DispatchQueue.main.async {
```

Comment:
I don't understand why you want to dispatch to the main queue here. I think most of the time you want handle networking in background queues.

Reply:
I just wanted to let current source be cancelled, but maybe I should remove this

Reply:
The "async" is needed, without it (easily reproduced by locking the device, and then unlocking it):
"recvfrom failed: Socket is not connected"
"Couldn\'t bind socket"

Reply:
closing because my last commit outdate this, waiting your review of this commit